backend/src/controllers/RestaurantController.ts

```diff
@@ -0,0 +1,67 @@
+import { Request, Response } from 'express';
+import Restaurant from '../models/restaurant';
+
+//http://localhost:3000/restaurants/search?searchQuery=pizza&selectedCuisines=Italian,Mexican&sortOption=restaurantName&page=2
+const searchRestaurants = async (req: Request, res: Response) => {
+  try {
+    const city = req.params.city;
+    const searchQuery = (req.query.searchQuery as string) || '';
+    const selectedCuisines = (req.query.selectedCuisines as string) || '';
+    const sortOption = (req.query.sortOption as string) || 'lastUpdated';
+    const page = parseInt(req.query.page as string) || 1;
+
+    const query: any = {};
+    query['city'] = new RegExp(city, 'i');
+    const cityCheck = await Restaurant.countDocuments(query);
+    if (cityCheck === 0) {
+      return res.status(404).json({
+        data: [],
+        pagination: {
+          total: 0,
+          page: 1,
+          pages: 1,
+        },
+      });
+    }
+
+    if (selectedCuisines) {
+      const cuisinesArray = selectedCuisines.split(',').map((cuisine) => new RegExp(cuisine, 'i'));
+      query['cuisines'] = { $all: cuisinesArray };
+    }
+
+    if (searchQuery) {
+      const searchRegex = new RegExp(searchQuery, 'i');
+      query['$or'] = [{ restaurantName: searchRegex }, { cuisines: { $in: [searchRegex] } }];
+    }
+
+    const pageSize = 10;
+    const skip = (page - 1) * pageSize;
+
+    // sortOption = "lastUpdated"
+    const restaurants = await Restaurant.find(query)
+      .sort({ [sortOption]: 1 })
+      .skip(skip)
+      .limit(pageSize)
+      .lean();
+
+    const total = await Restaurant.countDocuments(query);
+
+    const response = {
+      data: restaurants,
+      pagination: {
+        total,
+        page,
+        pages: Math.ceil(total / pageSize),
+      },
+    };
+
+    res.json(response);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({ message: 'Error searching restaurants' });
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/35#issuecomment-2406848249>#e519ab</a><span> - 10/11/2024, 08:13 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The error handling in the catch block logs the entire error object, which might contain sensitive information, and returns a generic error message to the client.
</div>
<div id="fix">
<b>Fix:</b> Implement proper error logging that excludes sensitive information. Provide more informative error messages to the client without revealing system details.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -59,4 +59,7 @@
    } catch (error) {
 -    console.log(error);
 -    res.status(500).json({ message: 'Error searching restaurants' });
 +    console.error('Error in searchRestaurants:', error.message);
 +    res.status(500).json({
 +      message: 'An error occurred while searching restaurants. Please try again later.'
 +    });
    }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged